### PR TITLE
[DOCS] Adds a signpost for downloading ES or signing-up for ESS

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a signpost to the repo README that links to the ES download page and to the ESS sign-up page.

### CODE REVIEW

**Review by pquentin — CHANGES_REQUESTED** [review]
The README file is (unfortunately) ReST, not Markdown, see https://github.com/elastic/elasticsearch-py/blob/bfa8db26287b5c961a01e7445552a716a5696995/README.rst for the current result.

File: README.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
    <img align="right" width="auto" height="auto" src="https://www.elastic.co/static-res/images/elastic-logo-200.png">
 
+**`Download the latest version of Elasticsearch <https://www.elastic.co/downloads/elasticsearch>`_**
+or
+**`sign-up <https://cloud.elastic.co/registration?elektra=en-ess-sign-up-page>`_**
+**for a free trial of Elastic Cloud**.
```
Comment (szabosteve) [inline README.rst:8]:
```suggestion
`**Download the latest version of Elasticsearch** <https://www.elastic.co/downloads/elasticsearch>`_
or
`**sign-up** <https://cloud.elastic.co/registration?elektra=en-ess-sign-up-page>`_
**for a free trial of Elastic Cloud**.
```

File: README.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
    <img align="right" width="auto" height="auto" src="https://www.elastic.co/static-res/images/elastic-logo-200.png">
 
+`**Download the latest version of Elasticsearch** <https://www.elastic.co/downloads/elasticsearch>`_
+or
+`**sign-up** <https://cloud.elastic.co/registration?elektra=en-ess-sign-up-page>`_
+**for a free trial of Elastic Cloud**.
```
Comment (szabosteve) [inline README.rst:8]:
```suggestion
`Download the latest version of Elasticsearch <https://www.elastic.co/downloads/elasticsearch>`_
or
`sign-up <https://cloud.elastic.co/registration?elektra=en-ess-sign-up-page>`_
for a free trial of Elastic Cloud.
```

**Comment by szabosteve** [comment]
@pquentin I addressed your comment via the last commits.

**Review by pquentin — COMMENTED** [review]
Thanks! What are your thoughts on moving those to the installation sections?

 * README.rst: either remove this or put it just before "Refer to the Installation section"
 * guide: either in https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/getting-started-python.html#_installation or in the installation page itself.
 * sphinx: I think this one works :+1:

**Comment by pquentin** [comment]
buildkite test this please

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Pull request merged**